core/src/main/scala/com.snowplowanalytics.elasticsearch/loader/NsqSourceExecutor.scala

```diff
@@ -106,19 +103,24 @@ class NsqSourceExecutor(
     }
 
     val errorCallback = new NSQErrorCallback {
-      override def error(e: NSQException): Unit = 
-        log.error(s"Exception while consuming topic $topicName", e)
+      override def error(e: NSQException): Unit =
+        log.error(s"Exception while consuming topic ${config.streams.inStreamName}", e)
     }
 
     // use NSQLookupd
     val lookup = new DefaultNSQLookup
-    lookup.addLookupAddress(config.nsq.host, config.nsq.lookupPort)
-    val consumer = new NSQConsumer(lookup,
-                                   topicName,
-                                   channelName,
-                                   nsqCallback,
-                                   new NSQConfig(),
-                                   errorCallback)
-    consumer.start() 
-  }   
+    config.queue match {
+      case queue: Nsq =>
+        lookup.addLookupAddress(queue.host, queue.lookupPort)
+        val consumer = new NSQConsumer(lookup,
+          config.streams.inStreamName,
+          queue.channelName,
+          nsqCallback,
+          new NSQConfig(),
+          errorCallback)
+        consumer.start()
+      case _ => throw new IllegalArgumentException("queue config is not valid for Nsq")
+    }
```

Comment:
this has already been validated so you should be able to directly pass a queue config which has type `Nsq`